homeassistant/components/anglian_water/__init__.py

```diff
@@ -0,0 +1,61 @@
+"""Anglian Water (UK) integration."""
+
+from __future__ import annotations
+
+from pyanglianwater import API, AnglianWater
+from pyanglianwater.exceptions import ServiceUnavailableError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_DEVICE_ID, DOMAIN
+from .coordinator import AnglianWaterDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [
+    Platform.SENSOR,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+    try:
+        _api = await API.create_via_login_existing_device(
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            entry.data[CONF_DEVICE_ID],
+        )
+        _aw = AnglianWater()
+        _aw.api = _api
+        # for future:
+        _aw.current_tariff = "not_set"
+        _aw.current_tariff_rate = 0.0
+
+        hass.data.setdefault(DOMAIN, {})
+        hass.data[DOMAIN][entry.entry_id] = coordinator = (
+            AnglianWaterDataUpdateCoordinator(hass=hass, client=_aw)
+        )
```

Comment:
Let's use config entry typing

Reply:
is there an example you can point me to please?

Reply:
https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry

When I was writing the review I was still on my phone so couldn't give an example, thanks for asking

Reply:
I like, this is pretty neat, something I didn't know about until now, thank you :)